packages/commerce-server/src/determine-purchase-type.ts

```diff
@@ -33,6 +33,98 @@ const match = (
   return Boolean(value1 && value2 && value1 === value2)
 }
 
+const {getPurchasesForUser, getPurchaseForStripeCharge} = getSdk()
+
+type PurchaseForStripeCharge = NonNullable<
+  Awaited<ReturnType<typeof getPurchaseForStripeCharge>>
+>
+type Purchase = Awaited<ReturnType<typeof getPurchasesForUser>>[number]
+type BulkCoupon = NonNullable<Pick<Purchase, 'bulkCoupon'>['bulkCoupon']>
+type BulkPurchase = Purchase & {bulkCoupon: BulkCoupon}
+type BulkPurchaseData = {
+  focalProduct: {
+    productId: string
+    focalPurchase: PurchaseForStripeCharge
+    otherTeamBulkPurchases: BulkPurchase[]
+    otherIndividualPurchases: Purchase[]
+    focalBulkPurchaseIsChronologicallyFirst: boolean
+  }
+}
+type IndividualPurchaseData = {
+  focalProduct: {
+    productId: string
+    focalPurchase: PurchaseForStripeCharge
+    otherIndividualPurchases: Purchase[]
+  }
+}
+
+const summarizePurchases = (
+  focalPurchase: PurchaseForStripeCharge,
+  allUserPurchases: Purchase[],
+):
+  | {type: 'BULK'; data: BulkPurchaseData}
+  | {type: 'INDIVIDUAL'; data: IndividualPurchaseData} => {
+  // summarize only purchases whose productId's match the focal
+  // purchase's productId
+  const focalProductId = focalPurchase.productId
+
+  const purchaseIsBulk = Boolean(focalPurchase.bulkCoupon?.id)
+
+  // Summarize the purchases for this product ID
+  const focalProductPurchases = allUserPurchases.filter(
+    (purchase) => purchase.productId === focalProductId,
+  )
```

Comment:
affordances for an app that may have multiple products that can be purchased

for now, we just focus on the product/`productId` that matches the current purchase's product and summarize based on that. We can easily extend this to summarize details of purchases for other products.

Reply:
need to rethink this part slightly -- this won't quite work with something like TA where there are 3 different products that correspond to different purchase tiers of the TA product.